libraries/Adafruit_LittleFS/src/Adafruit_LittleFS_File.h

```diff
@@ -69,13 +75,21 @@ class File : public Stream
     uint32_t size (void);
 
     void close (void);
+
+    __attribute__((deprecated("Recommend use of isOpen() for clarity")))
     operator bool (void);
+
+    bool isOpen(void);
     char const* name (void);
 
     bool isDirectory (void);
     File openNextFile (uint8_t mode = FILE_O_READ);
     void rewindDirectory (void);
 
+    // Has to be public, in order to grant 'friend' permissions to Adafruit_LittleFS private member _mutex
+    void DoNotCallFromOutsideClass_LockFilesystem(void);
```

Comment:
@henrygab  Do we need to lock the entire file system when working with File API, I am thinking we could have an mutex for each file/folder to ensure that there are 2 threads that open the same file/directory instead of locking the whole file system.